app/uk/gov/hmrc/transitmovements/repositories/DeparturesRepository.scala

```diff
@@ -120,15 +134,26 @@ class DeparturesRepositoryImpl @Inject() (
     })
   }
 
-  def getDepartureMessageIds(eoriNumber: EORINumber, departureId: DepartureId): EitherT[Future, MongoError, Option[NonEmptyList[MessageId]]] = {
+  def getDepartureMessageIds(
+    eoriNumber: EORINumber,
+    departureId: DepartureId,
+    receivedSince: Option[OffsetDateTime]
+  ): EitherT[Future, MongoError, Option[NonEmptyList[MessageId]]] = {
+
+    val selector = mAnd(
+      mEq("_id", departureId.value),
+      mEq("enrollmentEORINumber", eoriNumber.value)
+    )
 
-    val selector = mAnd(mEq("_id", departureId.value), mEq("enrollmentEORINumber", eoriNumber.value))
+    val dateTimeSelector = mGte("messages.received", receivedSince.map(_.toLocalDateTime).getOrElse(EPOCH_TIME))
```

Comment:
Would EPOCH_TIME potentially result in many messages being returned? Are messages actually purged from the database after 31 days? If so, I guess this will never be a problem.

Reply:
It'll return all messages for a given departure - however many that will be, and it'll only be IDs. There could be triple figures of messages on a departure in theory, but we're (thanfully) not returning all the contents of those, that could get rather large.

Messages are only purged when the departure itself is purged, which is 31 days after the last update to the departure (which will invariably be because a message was added to the departure).